lib/components/LoadingButton.tsx

```diff
@@ -1,24 +1,26 @@
-import { tx } from '../twind'
-import { Helpwave } from '../icons/Helpwave'
-import type { ButtonProps } from './Button'
-import { Button } from './Button'
+import clsx from 'clsx'
+import { Helpwave } from './icons/Helpwave'
+import type { SolidButtonProps } from './Button'
+import { ButtonSizePaddings } from './Button'
+import { SolidButton } from './Button'
 
 type LoadingButtonProps = {
   isLoading?: boolean,
-} & ButtonProps
+} & SolidButtonProps
+
+export const LoadingButton = ({ isLoading = false, size = 'medium', onClick, ...rest }: LoadingButtonProps) => {
+  const paddingClass = ButtonSizePaddings[size]
 
-export const LoadingButton = ({ isLoading = false, ...rest }: LoadingButtonProps) => {
   return (
-    <div className={tx('inline-block relative')}>
+    <div className={clsx('inline-block relative')}>
       {
         isLoading && (
-          <div className={tx('absolute inset-0 flex justify-center items-center bg-white/50')}>
-            <div className={tx('bg-white opacity-50')}/>
-            <Helpwave animate="loading"/>
+          <div className={clsx('absolute inset-0 row items-center justify-center bg-white/40', paddingClass)}>
+            <Helpwave animate="loading" className="text-white"/>
           </div>
         )
       }
-      <Button {...rest} disabled={rest.disabled || isLoading}/>
+      <SolidButton {...rest} disabled={rest.disabled} onClick={isLoading ? undefined: onClick}/>
```

Comment:
Should be disabled on click? Is there a logoc passing the `undefined` onClick action to disabled attribute - would be a good option I guess.